### PR TITLE
Support dropping schema in memory connector

### PR DESCRIPTION
**Pull request by kokosing**
Support dropping schema in memory connector

### CODE REVIEW

File: presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java [hunk 1]
```diff
@@ -119,6 +119,9 @@ public void testCreateTableInNonDefaultSchema()
 
         assertQueryResult(format("SELECT count(*) FROM %s.schema1.nation", CATALOG), 13L);
         assertQueryResult(format("SELECT count(*) FROM %s.schema2.nation", CATALOG), 12L);
+
+        assertUpdate(format("DROP SCHEMA %s.schema1", CATALOG));
+        assertUpdate(format("DROP SCHEMA %s.schema2", CATALOG));
```
Comment (findepi) [inline presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java]: \+ `assertUpdate(format("DROP SCHEMA IF EXISTS %s.schema2", CATALOG));`

Comment (findepi) [reply presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java]: although having a dedicated test for create/drop schema would be even better, covering dropping schemas with and without tables

File: presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryMetadata.java [hunk 2]
```diff
@@ -102,6 +104,20 @@ public synchronized void createSchema(ConnectorSession session, String schemaNam
...
+    @Override
+    public synchronized void dropSchema(ConnectorSession session, String schemaName)
+    {
+        if (!schemas.contains(schemaName)) {
+            throw new PrestoException(NOT_FOUND, format("Schema [%s] does not exist", schemaName));
+        }
+
+        tables.values().stream()
+                .filter(table -> table.getSchemaName().equals(schemaName))
+                .forEach(table -> dropTable(session, table));
```
Comment (findepi) [inline presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryMetadata.java]: Should it be allowed to drop schema with tables?

Comment (kokosing) [reply presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryMetadata.java]: Good point. It does not look as safe thing to do. Thanks.

**Comment by kokosing** [comment]
Applied comments

**Pull request merged**